File: server-kotlin-spring/src/main/kotlin/org/openapitools/model/RankListResponse.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonValue
import org.openapitools.model.RankResponse
import javax.validation.constraints.DecimalMax
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.Email
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size
import javax.validation.Valid
import io.swagger.v3.oas.annotations.media.Schema

/**
 * 
 * @param items 
 * @param userRank 
 * @param hasMoreResults 
 * @param count 
 * @param rankType 
 * @param app 
 * @param sortField 
 * @param leaderboardMode 
 * @param start 
 * @param limit 
 */
data class RankListResponse(

    @field:Valid
    @Schema(example = "null", description = "")
    @get:JsonProperty("items") val items: kotlin.collections.List<RankResponse>? = null,

    @field:Valid
    @Schema(example = "null", description = "")
    @get:JsonProperty("userRank") val userRank: RankResponse? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("hasMoreResults") val hasMoreResults: kotlin.Boolean? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("count") val count: kotlin.Int? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("rankType") val rankType: kotlin.String? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("app") val app: kotlin.String? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("sortField") val sortField: kotlin.String? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("leaderboardMode") val leaderboardMode: RankListResponse.LeaderboardMode? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("start") val start: kotlin.Long? = null,

    @Schema(example = "null", description = "")
    @get:JsonProperty("limit") val limit: kotlin.Long? = null
) {

    /**
    * 
    * Values: GLOBAL,FRIENDS,LOCAL,SEARCH,CUSTOM
    */
    enum class LeaderboardMode(@get:JsonValue val value: kotlin.String) {

        GLOBAL("GLOBAL"),
        FRIENDS("FRIENDS"),
        LOCAL("LOCAL"),
        SEARCH("SEARCH"),
        CUSTOM("CUSTOM");

        companion object {
            @JvmStatic
            @JsonCreator
            fun forValue(value: kotlin.String): LeaderboardMode {
                return values().firstOrNull{it -> it.value == value}
                    ?: throw IllegalArgumentException("Unexpected value '$value' for enum 'RankListResponse'")
            }
        }
    }

}

File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/RankListResponse.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class RankListResponse 
{
    public List<RankResponse> Items { get; set; }
    public RankResponse UserRank { get; set; }
    public bool HasMoreResults { get; set; }
    public int Count { get; set; }
    public string RankType { get; set; }
    public string App { get; set; }
    public string SortField { get; set; }
    
    /// <summary>
    /// Gets or Sets LeaderboardMode
    /// </summary>
    public enum LeaderboardModeEnum
    {
        
        /// <summary>
        /// Enum GLOBALEnum for GLOBAL
        /// </summary>
        GLOBALEnum = 1,
        
        /// <summary>
        /// Enum FRIENDSEnum for FRIENDS
        /// </summary>
        FRIENDSEnum = 2,
        
        /// <summary>
        /// Enum LOCALEnum for LOCAL
        /// </summary>
        LOCALEnum = 3,
        
        /// <summary>
        /// Enum SEARCHEnum for SEARCH
        /// </summary>
        SEARCHEnum = 4,
        
        /// <summary>
        /// Enum CUSTOMEnum for CUSTOM
        /// </summary>
        CUSTOMEnum = 5
    }

    public LeaderboardModeEnum LeaderboardMode { get; set; }
    public long Start { get; set; }
    public long Limit { get; set; }
}


